src/main/java/com/xceptance/xlt/report/providers/CustomLogsReportProvider.java

```diff
@@ -0,0 +1,217 @@
+     */
+    @Override
+    public void processDataRecord(final Data data)
+    {
+        // nothing to do here
+    }
+
+    /**
+     * Tell the system that there is no need to call processDataRecord
+     */
+    @Override
+    public boolean wantsDataRecords()
+    {
+        return false;
+    }        
+
+    /**
+     * Recursive method to walk the directory tree searching for custom log files. 
+     * Found files are directly put into a zip archive for the corresponding custom data scope in the report.
+     * @param file the directory or file to start from
+     * @throws IOException
+     */
+    private void findLogs (FileObject file) throws IOException
+    {
+        if (file.getType() == FileType.FOLDER && !XltConstants.CONFIG_DIR_NAME.equals(file.getName().getBaseName().toString()))
+        {
+            for (final FileObject fo : file.getChildren())
+            {
+                findLogs(fo);
+            }
+        }
+        else if (file.getType() == FileType.FILE)
+        {
+            if (file.getName().getBaseName().toString().startsWith(XltConstants.CUSTOM_LOG_PREFIX)) 
+            {
+                final String scopeName = file.getName().getBaseName().toString().substring(XltConstants.CUSTOM_LOG_PREFIX.length(), file.getName().getBaseName().toString().lastIndexOf('.'));
+                
+                if (foundScopes.isEmpty())
+                {
+                    //for the very first scope we have to create the directory to contain all logs in the report
+                    targetDir.toFile().mkdirs();
+                }
+                
+                // if scope has already been used, there is a stream for it
+                ZipOutputStream scopeStream = foundScopes.get(scopeName);
+                // if scope/stream does not exist yet, create one and add it to list
+                if (scopeStream == null)
+                {
+                    FileOutputStream fos = new FileOutputStream(targetDir.toString() + File.separator + scopeName + ".zip");
+                    scopeStream = new ZipOutputStream(fos);
+                    foundScopes.put(scopeName, scopeStream);
+                }
+                
+                // add zip entry, copy current log file for scope    
+                scopeStream.putNextEntry(new ZipEntry(baseDir.relativize(file.getPath()).toString().replaceAll("\\\\", "/")));
+                writeDataToZip(file, scopeStream);
+                scopeStream.closeEntry();
+            }
+        }
+        return;
+    }
+
+    private void writeDataToZip(FileObject file, ZipOutputStream scopeStream) throws IOException
+    {
+        final boolean isCompressed = "gz".equalsIgnoreCase(file.getName().getExtension());
+        
+        InputStream in = isCompressed ? 
+            new GZIPInputStream(file.getContent().getInputStream(), 1024 * 16) : //TODO is my data EVER zipped? it should be, but how does it happen?
+            file.getContent().getInputStream();
```

Comment:
Custom data is not gzipped at the moment, so we can remove this code.

Reply:
I thought the overall goal was to have all data zipped, so instead if removing this we should get the custom data zipped?